client/changelist/change.go

```diff
@@ -0,0 +1,73 @@
+package changelist
+
+import (
+	"encoding/json"
+
+	"github.com/endophage/gotuf/data"
+)
+
+type tufChange struct {
+	Actn       string `json:"action"`
+	Role       string `json:"role"`
+	ChangeType string `json:"type"`
+	ChangePath string `json:"path"`
+	Data       []byte `json:"data"`
+}
+
+func NewTufChange(action, role, changeType, changePath string, content []byte) *tufChange {
+	return &tufChange{
+		Actn:       action,
+		Role:       role,
+		ChangeType: changeType,
+		ChangePath: changePath,
+		Data:       content,
+	}
+}
+
+func NewTufTargetChange(action, target string, hash map[string]data.HexBytes, size int64, custom map[string]interface{}) (*tufChange, error) {
+	jsonCustom, err := json.Marshal(custom)
+	if err != nil {
+		return nil, err
+	}
+	customRaw := json.RawMessage{}
+	err = customRaw.UnmarshalJSON(jsonCustom)
+	if err != nil {
+		return nil, err
+	}
+	content := &data.FileMeta{
+		Length: size,
+		Hashes: hash,
+		Custom: &customRaw,
```

Comment:
Upstream as you say. I don't have a good reason, but the original flynn go-tuf implementation would let you add many targets with a single, shared piece of custom data in one operation so that may have been the original reason.